### DISCUSSION TITLE
it/guides/sahi-tiled-inference/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#18312] [Docs]
# it/guides/sahi-tiled-inference/

Imparate a implementare YOLO11 con SAHI per l'inferenza a fette. Ottimizzare l'uso della memoria e migliorare l'accuratezza del rilevamento per applicazioni su larga scala.

https://docs.ultralytics.com/it/guides/sahi-tiled-inference/

**Comment by davidetedd9203** [comment]
what are the differences between sahi and supervision inferenceSlicer?

**Reply by glenn-jocher** [reply]
@davidetedd9203 sAHI (Slicing Aided Hyper Inference) focuses on slicing large images into manageable parts for efficient object detection, optimizing memory usage and maintaining accuracy by stitching overlapping results. Supervision's `InferenceSlicer` provides a similar slicing mechanism but is tailored for integration into the Supervision library's specific workflows. For SAHI details, see [SAHI Docs](https://docs.ultralytics.com/guides/sahi-tiled-inference/).

**Comment by UltralyticsAssistant** [comment]
👋 Ciao, grazie per il tuo interesse in Ultralytics e il nuovo YOLO 🚀! Ti consigliamo di dare un'occhiata alla nostra [Documentazione](https://docs.ultralytics.com) per una raccolta completa di esempi e guide, incluso [l'uso di Python](https://docs.ultralytics.com/usage/python/) e [l'interfaccia a riga di comando (CLI)](https://docs.ultralytics.com/usage/cli/), dove molte domande comuni trovano risposta.

Se questa è una 🐛 segnalazione di bug, forniscici un [esempio minimo riproducibile (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) per aiutarci a diagnosticare il problema.

Se si tratta di una domanda ❓ su come personalizzare o addestrare un modello, per favore condividi quante più informazioni possibili, inclusi esempi di immagini del dataset e i log di addestramento, e verifica che tu stia seguendo i nostri [Suggerimenti per i Migliori Risultati di Addestramento](https://docs.ultralytics.com/guides/model-training-tips/).

Unisciti alla comunità Ultralytics dove preferisci! Per domande in tempo reale, visita il nostro [Discord](https://discord.com/invite/ultralytics) 🎧, o per discussioni approfondite ti invitiamo su [Discourse](https://community.ultralytics.com). Vuoi condividere o esplorare con il nostro subreddit? Prova [Reddit](https://reddit.com/r/Ultralytics) 📚.

## Upgrade

Aggiorna alla versione più recente del pacchetto `ultralytics` con tutti i [requisiti](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in un ambiente [**Python>=3.8**](https://www.python.org/) e [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), per verificare se il problema è già stato risolto:

```bash
pip install -U ultralytics
```

## Ambienti

Puoi eseguire YOLO in uno dei seguenti ambienti aggiornati e verificati (con tutte le dipendenze incluse [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) e [PyTorch](https://pytorch.org/) preinstallati):

- **Notebook** con GPU gratuita: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Consulta la nostra [Guida Rapida GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **AMI Deep Learning Amazon**. Consulta la nostra [Guida AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Immagine Docker**. Consulta la nostra [Guida Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Stato

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Se questo badge è verde, tutti i test [CI di Ultralytics](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) sono attualmente superati. I test CI verificano il corretto funzionamento di tutte le [Modalità](https://docs.ultralytics.com/modes/) e [Compiti](https://docs.ultralytics.com/tasks/) di YOLO su macOS, Windows e Ubuntu quotidianamente e per ogni commit.

Questo è un messaggio automatico 🤖. Un ingegnere Ultralytics ti assisterà presto. 😊

**Comment by MasIgor** [comment]
Hello! Could you please elaborate on how the training would work? Lets say i have 100 images with 24 mpx. what would the steps be? would I label the 24 mpx images and use sahi for training? if yes, how? or would i have to extract 640x640 images on my own from the 24mpx images and do the training as usual? 
any help is greatly appreciated.

thanks
